pandas/plotting/_timeseries.py

```diff
@@ -0,0 +1,339 @@
+# TODO: Use the fact that axis can have units to simplify the process
+
+import numpy as np
+
+from matplotlib import pylab
+from pandas.tseries.period import Period
+from pandas.tseries.offsets import DateOffset
+import pandas.tseries.frequencies as frequencies
+from pandas.tseries.index import DatetimeIndex
+from pandas.tseries.period import PeriodIndex
+from pandas.tseries.tdi import TimedeltaIndex
+from pandas.formats.printing import pprint_thing
+import pandas.compat as compat
+
+from pandas.plotting._converter import (TimeSeries_DateLocator,
+                                        TimeSeries_DateFormatter,
+                                        TimeSeries_TimedeltaFormatter)
+
+# ---------------------------------------------------------------------
+# Plotting functions and monkey patches
+
+
+def tsplot(series, plotf, ax=None, **kwargs):
+    """
+    Plots a Series on the given Matplotlib axes or the current axes
+
+    Parameters
+    ----------
+    axes : Axes
+    series : Series
+
+    Notes
+    _____
+    Supports same kwargs as Axes.plot
+
+    """
+    # Used inferred freq is possible, need a test case for inferred
+    if ax is None:
+        import matplotlib.pyplot as plt
+        ax = plt.gca()
+
+    freq, series = _maybe_resample(series, ax, kwargs)
+
+    # Set ax with freq info
+    _decorate_axes(ax, freq, kwargs)
+    ax._plot_data.append((series, plotf, kwargs))
+    lines = plotf(ax, series.index._mpl_repr(), series.values, **kwargs)
+
+    # set date formatter, locators and rescale limits
+    format_dateaxis(ax, ax.freq, series.index)
+    return lines
+
+
+def _maybe_resample(series, ax, kwargs):
```

Comment:
side issue since this entire module is private, the functions themselves don't need to be (but can fix later)